connectors/byoc.py

```diff
@@ -160,13 +160,15 @@ def _create_object(self, doc_source):
 
 
 class SyncJob:
-    def __init__(self, connector_id, elastic_client):
+    def __init__(self, elastic_index, connector_id, doc_source=None):
```

Comment:
I don't understand why `SyncJob` holds `doc_source` I am not sure how and where it's used since it can be `None`

Shouldn't it be initialized to an empty dict?

I am wondering if `SyncJob` (and maybe other classes) should not inherit from an `ESDocument` base class that deals with `doc_source`?

Reply:
Agree it shouldn't accept `None`. 

> I am wondering if `SyncJob` (and maybe other classes) should not inherit from an `ESDocument` base class that deals with `doc_source`

It makes a lot of sense to have a `ESDocument` class but I think it's still in draft mode: https://github.com/elastic/connectors-python/pull/213

Reply:
>I don't understand why SyncJob holds doc_source I am not sure how and where it's used since it can be None

I implemented it the same as I did for `Connector` which also holds `doc_source`.  There will be more refactoring which should touch `SyncJob` so `doc_source=None` is a mock for now. The idea behind this is: `ESDocument` contains all the logic required to create/update/delete documents safely using `Optimistic concurrency control`. Also, `SyncJob` and `Connector` might have some common logic and use `ESDocument` as a provider in case we have to use Mysql/Postgres. 

I think it makes sense to use `dict()` instead of None: